framework/doc/content/source/problems/EigenProblem.md

```diff
@@ -2,16 +2,109 @@
+ neutron transport with temperature feedback. In this case,
+ $A$ and $B$ are not constant anymore; instead, they depend on
+ the eigenvector directly or indirectly. To describe our nonlinear
+ solver, we write a generalized nonlinear eigenvalue problem as,
+\begin{equation}
+A(x) = \lambda B(x),
+\end{equation}
+where $A(\cdot)$ and $B$ are nonlinear functions. In the EigenProblem
+system, the kernels of $A(\cdot)$ are referred to as "noneigen,"
+and the kernels of $B(\cdot)$ are referred to as "eigen."
+To utilize Newton to solve this problem, we define $\lambda = \frac{1}{|B(x)|}$,
+and rewrite the nonlinear eigen problem as
+\begin{equation}
+0=F(x) = A(x) - \lambda B(x).
+\end{equation}
+Newton can be applied to solve this problem. The only issue is that
+the derivatives of $\lambda$ with respective to $x$ can be dense.
+We use the Jacobian-free Newton method to overcome this difficulty,
+where the action of the Jacobian is approximated via finite differences.
+To handle various use cases, variants of Newton are provided
+in this system. There is a current list:
+
+- `PJFNK` - Preconditioned Jacobian-free Newton Krylov: The preconditioning matrix
+  is formed using noneigen kernels. We do have an option to allow users to
+  incorporate eigen kernels into the preconditioning matrix. Still, in
+  general, we won't encourage users to do that since the preconditioning
+  matrix might be singular when close to the solution.
+
+!listing test/tests/problems/eigen_problem/eigensolvers/ne.i
+          block=Executioner
+          link=False
+
+- `JFNK` - Jacobian-free Newton Krylov:  Users are responsible for
+  providing a preconditioner. This allows users to build a customized
+  preconditioner, such as a sweeper,
+  that might be more efficient for their underlying problems.
+
+!listing test/tests/problems/eigen_problem/preconditioners/ne_pbp.i
+          block=Executioner Preconditioning
+          link=False
+
+- `PJFNKMO`- Matrix-only Preconditioned Jacobian-free Newton Krylov: Mathematically,
+ this option is the same as PJFNK. The difference is the residual evaluation.
+ PJFNK calls the user residual/function evaluation routine, which includes all finite element calculations.
+ PJFNKMO forms residual vectors via  matrix-vector multiplications: $A (x) = Ax$
+ and $B(x) = Bx$. This is useful when matrices are constant. This option will be
+ more efficient. If the matrices are not constant, please do not use this option;
+ otherwise, simulation results will be wrong.
+
+!listing test/tests/problems/eigen_problem/jfnk_mo/ne_array_mo.i
+           block=Executioner
+           link=False
+
+- `Newton` - Newton method: Both Jacobian and preconditioning matrices are $A$.
+ This option is added for consistency between the nonlinear system and
+ the nonlinear eigensystem. Since this option does not account for the derivative
+ of eigenvalue with respective to the eigenvector, this option is not efficient.
+ Users should not use it in general.
```

Comment:
I think in this case, Newton iteration is essentially the inverse power iteration. If so, you may want to add this.

Reply:
This is a good point. The observation  inspired me think more  on the inverse power iteration.  I am not sure this case is  the inverse power iteration. I think it is very different from the inverse power iteration.   I need to do math to figure it out.

But at this point, let us do not add that conclusion .

Reply:
ok to me. If we are doing this `x = xold + A^-1 * residual`, we are doing inverse power iteration.